woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php

```diff
@@ -418,9 +423,15 @@ public function get_payment_terms($amount, $max_instalments, $tax = 0) {
 	public function payment_fields() {
 		$cart_total = $this->get_order_total();
 
-		$cards = array_filter((array) get_user_meta($this->userId, '_ebanx_credit_card_token', true), function ($card) {
-			return !empty($card->brand) && !empty($card->token) && !empty($card->masked_number);
-		});
+		$cards = array();
+
+		$save_card = $this->get_setting_or_default('save_card_data', 'no');
+
+		if ( $save_card && $save_card === 'yes' ) {
```

Comment:
`$save_card` has a default value of "no", so it would be better to change line 428 to:
`$save_card = $this->get_setting_or_default('save_card_data', 'no') === 'yes';`
And then the if would only check for `$save_card`.